### PR TITLE
chore: add strict eslint rules

### PR DESCRIPTION
**Pull request by IrakliJani**
Closes: #5336 

### Description:

adds strict checks on:

- [x] `react-hooks/exhaustive-deps`
- [x] `no-unused-vars`
- [x] `@typescript-eslint/no-unused-vars`

unused variables are ignored if:

- function argument starts with `_`: `array.filter(_elem, index)`
- array destructure variable starts with `_`: `[_first, second] = array`
- object destructure has a spread variable: `{ name, ...personWithoutName } = person`

- [x] removes eslint errors and fixes build

⚠️ only affects frontend builds

### CODE REVIEW

File: packages/frontend/.eslintrc.js [hunk 1]
```diff
@@ -37,6 +43,18 @@ module.exports = {
...
+        '@typescript-eslint/naming-convention': [
+            'error',
+            {
+                selector: 'variable',
+                format: ['camelCase', 'PascalCase', 'UPPER_CASE'],
+                leadingUnderscore: 'allow',
+            },
+        ],
+        'react-hooks/exhaustive-deps': 'error',
+
```
Comment (IrakliJani) [inline packages/frontend/.eslintrc.js:57]: config changes. below that is just removing unused variables

File: packages/frontend/src/types/table-core.d.ts [hunk 2]
```diff
@@ -1,10 +1,9 @@
 import { Field, PivotReference, TableCalculation } from '@lightdash/common';
-import { RowData } from '@tanstack/react-table';
 import { MouseEventHandler } from 'react';
 import { Sort } from '../components/common/Table/types';
 
 declare module '@tanstack/table-core' {
-    interface ColumnMeta<TData extends RowData, TValue> {
+    interface ColumnMeta {
```
Comment (joaoviana) [inline packages/frontend/src/types/table-core.d.ts:6]: not sure about this change, why was this removed here? 🤔

File: packages/frontend/src/pages/SavedDashboards.tsx [hunk 3]
```diff
@@ -28,13 +27,6 @@ const SavedDashboards = () => {
     const [isCreateDashboardOpen, setIsCreateDashboardOpen] =
         useState<boolean>(false);
 
-    const {
-        isLoading: isCreatingDashboard,
-        isSuccess: hasCreatedDashboard,
-        mutate: createDashboard,
-        data: newDashboard,
-    } = useCreateMutation(projectUuid);
-
```
Comment (joaoviana) [inline packages/frontend/src/pages/SavedDashboards.tsx:37]: hmm is this intended? don't we need this?

File: packages/frontend/src/hooks/user/useJoinOrganizationMutation.ts [hunk 4]
```diff
@@ -1,11 +1,7 @@
-import {
-    ApiError,
-    CreateOrganization,
-    UpdateOrganization,
-} from '@lightdash/common';
+import { ApiError } from '@lightdash/common';
 import { useMutation, useQueryClient } from 'react-query';
+
```
Comment (joaoviana) [inline packages/frontend/src/hooks/user/useJoinOrganizationMutation.ts:3]: how is this new line added? through linting?

File: packages/frontend/src/components/common/ResourceView/ResourceViewGrid/index.tsx [hunk 5]
```diff
@@ -47,9 +48,9 @@ const ResourceViewGrid: FC<ResourceViewGridProps> = ({
             .filter((group) => group.items.length > 0);
     }, [groups, items]);
 
-    const [draggableItems, setDraggableItems] = useState(
-        groupedItems.map((g) => g.items),
-    );
+    // const [draggableItems, setDraggableItems] = useState(
+    //     groupedItems.map((g) => g.items),
+    // );
```
Comment (joaoviana) [inline packages/frontend/src/components/common/ResourceView/ResourceViewGrid/index.tsx:53]: this can be removed then? A bit confused about some changes on this PR...

**Review by joaoviana — COMMENTED** [review]
Great stuff! Some Qs around the changes on this PR that need some clarity.

File: packages/frontend/src/types/table-core.d.ts (hunk 2 above)
Comment (IrakliJani) [reply packages/frontend/src/types/table-core.d.ts:6]: unused types

File: packages/frontend/src/hooks/user/useJoinOrganizationMutation.ts (hunk 4 above)
Comment (IrakliJani) [reply packages/frontend/src/hooks/user/useJoinOrganizationMutation.ts:3]: must have been me

File: packages/frontend/src/components/common/ResourceView/ResourceViewGrid/index.tsx (hunk 5 above)
Comment (IrakliJani) [reply packages/frontend/src/components/common/ResourceView/ResourceViewGrid/index.tsx:53]: @joaoviana pinning drag and drop items is work in progress. variable was unused. commented out to not remove code and fix the linting error

File: packages/frontend/src/pages/SavedDashboards.tsx (hunk 3 above)
Comment (joaoviana) [reply packages/frontend/src/pages/SavedDashboards.tsx:37]: will wait for confirmation and then approve this!

Comment (yasmin-e) [reply packages/frontend/src/pages/SavedDashboards.tsx:37]: yep, check out `packages/frontend/src/components/common/modal/DashboardCreateModal.tsx`

**Review by joaoviana — APPROVED** [review]
